idea/src/org/jetbrains/kotlin/idea/intentions/RemoveRedundantCallsOfConversionMethodsIntention.kt

```diff
@@ -0,0 +1,69 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.core.replaced
+import org.jetbrains.kotlin.idea.inspections.IntentionBasedInspection
+import org.jetbrains.kotlin.js.descriptorUtils.getJetTypeFqName
+import org.jetbrains.kotlin.psi.KtConstantExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtStringTemplateExpression
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import java.util.*
+
+class RemoveRedundantCallsOfConversionMethodsInspection : IntentionBasedInspection<KtDotQualifiedExpression>(RemoveRedundantCallsOfConversionMethodsIntention::class)
+
+class RemoveRedundantCallsOfConversionMethodsIntention : SelfTargetingOffsetIndependentIntention<KtDotQualifiedExpression>(KtDotQualifiedExpression::class.java, "Remove redundant calls of the conversion method") {
+
+    private val targetClassMap = mapOf("toList()" to List::class.qualifiedName,
+                                       "toSet()" to Set::class.qualifiedName,
+                                       "toMap()" to Map::class.qualifiedName,
+                                       "toMutableList()" to "kotlin.collections.MutableList",
+                                       "toMutableSet()" to "kotlin.collections.MutableSet",
+                                       "toMutableMap()" to "kotlin.collections.MutableMap",
+                                       "toSortedSet()" to SortedSet::class.qualifiedName,
+                                       "toSortedMap()" to SortedMap::class.qualifiedName,
+                                       "toString()" to String::class.qualifiedName,
+                                       "toDouble()" to Double::class.qualifiedName,
+                                       "toFloat()" to Float::class.qualifiedName,
+                                       "toLong()" to Long::class.qualifiedName,
+                                       "toInt()" to Int::class.qualifiedName,
+                                       "toChar()" to Char::class.qualifiedName,
+                                       "toShort()" to Short::class.qualifiedName,
+                                       "toByte()" to Byte::class.qualifiedName)
+
+
+    override fun applyTo(element: KtDotQualifiedExpression, editor: Editor?) {
+        element.replaced(element.receiverExpression)
+
+        mutableMapOf(1 to 1).toMutableMap()
+    }
+
+    override fun isApplicableTo(element: KtDotQualifiedExpression): Boolean {
```

Comment:
Looks like we can apply this intention for all qualified expressions, not only unsafe ones.